jenkins/deploy-clio-server.sh

```diff
@@ -170,59 +154,126 @@ render_ctmpls() {
   rm ${ctmpl_env_file}
 }
 
-stop_containers() {
-  local -r clio_fqdn=$1
+activate_service_account() {
+  VAULT_TOKEN=$(cat /etc/vault-token-dsde)
+  SA_VAULT_PATH="secret/dsde/gotc/dev/common/ci-deployer-service-account.json"
+
+  docker run --rm -e VAULT_TOKEN="${VAULT_TOKEN}" \
+    broadinstitute/dsde-toolbox:dev vault read -format=json ${SA_VAULT_PATH} \
+    | jq '.data' > service-account.json
 
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "test -f ${COMPOSE_FILE} && ${DOCKER_COMPOSE[*]} stop || echo"
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "test -f ${COMPOSE_FILE} && ${DOCKER_COMPOSE[*]} rm -f || echo"
+  # activate SA
+  gcloud auth activate-service-account --key-file=service-account.json
 }
 
-start_containers() {
-  local -r clio_fqdn=$1
+get_clio_instance() {
+  CLIO_PROJECT=$1
 
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "${DOCKER_COMPOSE[*]} pull"
-  ssh ${SSH_OPTS[@]} ${SSH_USER}@${clio_fqdn} "${DOCKER_COMPOSE[*]} up -d"
+  # Get the instance marked with the Clio and Active labels (There should only be one)
+  CLIO_INSTANCE_LIST=$(gcloud --format="table[no-heading](Name)" compute \
+                         --project ${CLIO_PROJECT} instances list \
+                         --filter="labels.app=clio AND labels.state=active")
+
+  # Verify that there is only one instance
+  if [[ ${#CLIO_INSTANCE_LIST[@]} -ne 1 ]];
```

Comment:
if you end the line with a `;`, then you can put the `then` after it, e.g. `if [[ ... ]]; then`. If you want to put `then` on its own line, then you don't need a `;` since it's a command separator and there isn't a command after it. Personally I prefer `; then` but either is OK.

Reply:
I know it's not needed, but the linter in my IDE liked it, so I left it.